src/stirling/source_connectors/socket_tracer/protocols/common/data_stream_buffer.h

```diff
@@ -95,7 +95,21 @@ class DataStreamBuffer {
    * @param pos The logical position of the data.
    * @return The timestamp or error if the position does not contain valid data.
    */
-  StatusOr<uint64_t> GetTimestamp(size_t pos) const { return impl_->GetTimestamp(pos); }
+  StatusOr<uint64_t> GetTimestamp(size_t pos) {
+    StatusOr<uint64_t> timestamp_ns_status = impl_->GetTimestamp(pos);
+    if (!timestamp_ns_status.ok()) {
+      return timestamp_ns_status;
+    }
+    uint64_t current_timestamp_ns = timestamp_ns_status.ValueOr(0);
+    uint64_t prev_timestamp_ns = GetPrevTimestamp().ValueOr(0);
```

Comment:
I'm open to other's thoughts, but I think it's cleaner here to use `ConsumeValueOrDie()`. The implementation of that function checks the same field as `StatusOr::ok()`, so it should be safe unless we feel that the object will be changed in between these calls.

@etep what do you think?